Editors and Overlays: remove deprecated options

aleksei-semikozov opened the pull request:
PR with .d.ts removing https://github.com/DevExpress/DevExtreme/pull/30189

- HtmlEditor: formatName and formatValues - [link](https://js.devexpress.com/jQuery/Documentation/ApiReference/UI_Components/dxHtmlEditor/Configuration/toolbar/items/#formatName) (21.2) - removed deprecation warnings
- Lookup [valueChangeEvent](https://js.devexpress.com/jQuery/Documentation/ApiReference/UI_Components/dxLookup/Configuration/#valueChangeEvent) (22.1)
- SelectBox [valueChangeEvent](https://js.devexpress.com/jQuery/Documentation/ApiReference/UI_Components/dxSelectBox/Configuration/#valueChangeEvent) (22.2)
- Overlays and ContextMenu [closeOnOutsideClick](https://js.devexpress.com/jQuery/Documentation/ApiReference/UI_Components/dxSelectBox/Configuration/#valueChangeEvent) (21.2)

Commit:
Lookup: remove fullScreen option d.ts

Commit:
Overlay & ContextMenu: remove closeOnOutsideClick option d.ts
affect components:
- LoadPanel
- ContextMenu
- Popover
- Popup
- Toast
- Tooltip

Commit:
Lookup & SelectBox: remove valueChangeEvent option d.ts

Commit:
HtmlEditor: remove formatName and formatValues option d.ts

Commit:
Overlays: remove elementAttr option d.ts

Commit:
Widget: remove height/width option as function deprecations. d.ts

Commit:
Roll back format mistake in /viz/core/base_widget.d.ts

Commit:
Revert "Overlays: remove elementAttr option d.ts"

This reverts commit 16528f4de11a8445a6256382437f16bbd8c9c8e9.

Commit:
fix & hide typescript errors

Commit:
Fix ts-lint error

Commit:
Context Menu: remove deprecated closeOnOutsideClick option

Commit:
Overlay: remove deprecated closeOnOutsideClick option

Commit:
Context Menu & Popover & Toast: remove deprecated closeOnOutsideClick option

Commit:
Lookup & SelectBox: remove deprecated valueChangeEvent option

Commit:
HtmlEditor: remove deprecated formatName and formatValues properties

Commit:
Merge branch '25_1' into depreacte_options_25_1_code

ksercs commented on packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/selectBox.tests.js:
```diff
@@ -6593,24 +6587,6 @@ QUnit.module('The "customItemCreateEvent" option warning', {
-        }
-    });
-
     QUnit.test('no warning should be logged on pure init', function(assert) {
```
looks like the whole module does not make sense

marker-dao commented on packages/devextreme/js/__internal/ui/context_menu/m_context_menu.ts:
```diff
@@ -630,15 +623,13 @@ class ContextMenu extends MenuBase {
   _shouldHideOnOutsideClick(e) {
-    // @ts-expect-error
-    const { closeOnOutsideClick, hideOnOutsideClick } = this.option();
+    const hideOnOutsideClick = this.option('hideOnOutsideClick');
```
**[minor]**
I would prefer not to use WA to avoid ts-error. It is better to leave a comment // @ts-expect-error so that attention can be drawn to this issue in the future.
```suggestion
    // @ts-expect-error hideOnOutsideClick does not exist on type 'Properties'
    const { hideOnOutsideClick } = this.option();
```

aleksei-semikozov replied:
I'm not sure, I did it for consistency with other parts of the code. Maybe your outline is actually better

marker-dao commented on packages/devextreme/testing/tests/DevExpress.ui.widgets/menu.tests.js:
```diff
@@ -1632,7 +1632,7 @@ QUnit.module('Menu tests', {
         assert.ok(submenu._overlay.option('visible'));
 
-        $(document).trigger('dxpointerdown'); // it needs to trigger closeOnOutsideClick
+        $(document).trigger('dxpointerdown'); // it needs to trigge hideOnOutsideClick
```
**[minor]**
```suggestion
        $(document).trigger('dxpointerdown'); // it needs to trigger hideOnOutsideClick
```

ksercs commented:
Missed points:
1. Warning for HtmlEditor: https://github.com/DevExpress/DevExtreme/blob/25_1/packages/devextreme/js/ui/widget/ui.errors.js#L322
2. Selectbox valueChangeEvent default value

aleksei-semikozov committed:
Update packages/devextreme/testing/tests/DevExpress.ui.widgets/menu.tests.js

Co-authored-by: marker dao ® <mark_nikolsky@outlook.com>
Signed-off-by: Aleksei Semikozov <alwexy@gmail.com>

Commit:
Fix after review

SelectBox: remove valueChangeEvent default option
SelectBox: remove redundancy test module
HtmlEditor: remove redundancy warning
